docs/reference/profile.rst

```diff
@@ -0,0 +1,19 @@
+.. _profile:
```

Comment:
Out of scope for this PR, but this whole page is probably a prime target for deletion and inclusion into [a glossary file](https://github.com/canonical/charmcraft/issues/2089)